pkg/solver/store/memory/store.go

```diff
@@ -197,6 +197,17 @@ func (s *SolverStoreMemory) GetResourceOffer(id string) (*data.ResourceOfferCont
 	return resourceOffer, nil
 }
 
+func (s *SolverStoreMemory) GetResourceOfferByAddress(address string) (*data.ResourceOfferContainer, error) {
+	s.mutex.RLock()
+	defer s.mutex.RUnlock()
+	for _, resourceOffer := range s.resourceOfferMap {
+		if resourceOffer.ResourceProvider == address {
```

Comment:
Do we want to do a toLowercase on these two strings just to be safe?

Reply:
better safe than sorry! done.